lib/cluster-providers/ec2-cluster-provider.ts

```diff
@@ -84,8 +85,17 @@ export class EC2ClusterProvider implements ClusterProvider {
     readonly options: EC2ProviderClusterProps;
 
 
-    constructor(options?: EC2ProviderClusterProps) {
-        this.options = options ?? { version: KubernetesVersion.V1_19 };
+    constructor(scope?: Construct, options?: EC2ProviderClusterProps) {
+        if (options != null)
+        {
+            this.options = options;
+        }
+        else {
+            if (scope != null)
+                this.options = { version: kubernetesVersionContext(scope)};
```

Comment:
This will just spread the scope for kubernetes version initialization from context, etc. Let's move the version into the options. We can also supply an implementation of EC2ProviderClusterProps like `DefaultEC2Options implements  EC2ProviderClusterProps` and centralize the default logic inside the class by initializing instance variables with defaults.

Reply:
Is this going to be an open issue to correct in the future?

Reply:
Ideally, I would not want to pass scope in every constructor to just initialize defaults. Please see my other comments on the subject. You could use the createCluster atm. We could defer the more holistic approach.